serverless/aws/lambda/src/main/scala/sttp/tapir/serverless/aws/lambda/runtime/AwsLambdaRuntimeLoop.scala

```diff
@@ -0,0 +1,103 @@
+
+    val runtimeApiInvocationUri = uri"http://${awsRuntimeApi}/2018-06-01/runtime/invocation"
+
+    /** Make request (without a timeout as prescribed by the AWS Custom Lambda Runtime documentation).
+      * This is due to the possibility of the runtime being frozen between lambda function invocations.
+      */
+    val nextEventRequest = basicRequest.get(uri"$runtimeApiInvocationUri/next").response(asStringAlways).readTimeout(0.seconds)
+
+    val pollEvent: F[RequestEvent] = {
+      logger.info(s"Fetching request event")
+      backend
+        .use(nextEventRequest.send(_))
+        .flatMap { response =>
+          response.header("lambda-runtime-aws-request-id") match {
+            case Some(id) => RequestEvent(id, response.body).unit
+            case _ =>
+              monad.error[RequestEvent](new RuntimeException(s"Missing lambda-runtime-aws-request-id header in request event $response"))
+          }
+        }
+        .handleError { case e => monad.error(new RuntimeException(s"Failed to fetch request event, ${e.getMessage}")) }
+    }
+
+    val decodeEvent: RequestEvent => F[AwsRequest] = event => {
+      decode[AwsRequest](event.body) match {
+        case Right(awsRequest) => awsRequest.unit
+        case Left(e)           => monad.error(new RuntimeException(s"Failed to decode request event ${event.requestId}, ${e.getMessage}"))
+      }
+    }
+
+    val routeRequest: (RequestEvent, AwsRequest) => F[Either[Throwable, AwsResponse]] = (event, request) =>
+      route(request).map(_.asRight[Throwable]).handleError { case e =>
+        logger.error(s"Failed to process request event ${event.requestId}", e)
+        e.asLeft[AwsResponse].unit
+      }
+
+    val sendResponse: (RequestEvent, AwsResponse) => F[Unit] = (event, response) =>
+      backend
+        .use { b =>
+          basicRequest
+            .post(uri"$runtimeApiInvocationUri/${event.requestId}/response")
+            .body(Printer.noSpaces.print(response.asJson))
+            .send(b)
+        }
+        .map(_ => ())
+        .handleError { case e => monad.error(new RuntimeException(s"Failed to send response for event ${event.requestId}")) }
+
+    val sendError: (RequestEvent, Throwable) => F[Unit] = (event, e) =>
+      backend
+        .use { b =>
+          basicRequest.post(uri"$runtimeApiInvocationUri/${event.requestId}/error").body(e.getMessage).send(b)
+        }
+        .map(_ => ())
+        .handleError { case e => monad.error(new RuntimeException(s"Failed to send error for event ${event.requestId}")) }
+
+    val sendResult: (RequestEvent, Either[Throwable, AwsResponse]) => F[Unit] = (event, result) =>
+      result match {
+        case Right(response) =>
+          logger.info(s"Request event ${event.requestId} completed successfully")
+          sendResponse(event, response)
+        case Left(e) =>
+          logger.error(s"Request event ${event.requestId} failed", e)
+          sendError(event, e)
+      }
+
+    (for {
```

Comment:
will this loop? :)

Reply:
nevermind ;)

Reply:
maybe it should be named `AwsLambdaRuntimeLogic` instead of loop